packages/backend/src/database/entities/notifications.ts

```diff
@@ -2,25 +2,49 @@ import { Knex } from 'knex';
 
 export const NotificationsTableName = 'notifications';
 
-export type DbNotifications = {
+enum NotificationResourceType {
+    Comment = 'comment',
+}
+
+interface NotificationDashboardTileCommentMetadata {
+    dashboard_uuid: string;
+    dashboard_name: string;
+    dashboard_tile_uuid: string;
+    dashboard_tile_name: string;
+}
+
+type NotificationMetadataTypes = {
+    [NotificationResourceType.Comment]: NotificationDashboardTileCommentMetadata;
+};
+
+type DbNotificationBase = {
     notification_id: string;
     created_at: Date;
     viewed: boolean;
     user_uuid: string;
-    // Dashboard tile comment-related fields
-    comment_id: string | null;
-    dashboard_uuid: string | null;
+    resource_uuid: string | null;
+    message: string | null;
+    url: string | null;
```

Comment:
Why not having resource_type in the base like `resource_type: 'comments'`  or `resource_type: NotificationType`

and `metadata: NotificationCommentMetadata | <other metadata types>`

I find this easier to read

Reply:
This provides better stricter type association between the `resource_type` and the `metadata`, so we don't need to rely on run-time checks. 
But, with readability in mind, I will move this DbNotificationBase to below so it's all in one place

Reply:
simplified a bit more: https://github.com/lightdash/lightdash/pull/9131/commits/8f7531d59983fc8962518dc17054c64b5f05e65d